src/app/public/modules/dynamic-component/dynamic-component.service.ts

```diff
@@ -86,11 +86,8 @@ export class SkyDynamicComponentService {
   public removeComponent<T>(
     cmpRef: ComponentRef<T>
   ): void {
-    const bodyEl = this.windowRef.getWindow().document.body;
-
-    const el = this.getRootNode(cmpRef);
-
-    this.renderer.removeChild(bodyEl, el);
+    this.appRef.detachView(cmpRef.hostView);
```

Comment:
This service method is public, which means it is expected that a consumer will run this in their SPAs. It would be good to make sure that a `cmpRef` is provided beforehand.
```
if (!cmpRef) {
  return;
}

this.appRef.detachView(cmpRef.hostView);
cmpRef.destroy();
```

Reply:
Done